walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,118 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*; 
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> mandatoryTimeRanges = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> optionalTimeRanges = new ArrayList<TimeRange>();
+    // if meeting duration is longer than a day, it is not possible to hold meeting
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return mandatoryTimeRanges;
+    }
+    // initialize time range with whole day
+    mandatoryTimeRanges.add(TimeRange.WHOLE_DAY);
+    optionalTimeRanges.add(TimeRange.WHOLE_DAY);
+    Collection<String> MandatoryAttendants = request.getAttendees();
+    Collection<String> optionalAttendants = request.getOptionalAttendees();
+    for(Event e: events) {
+      boolean checkMandatoryAttendee = MandatoryAttendants.size() == 0 ? false : true;
```

Comment:
The variable was intended to just ensure that I don't look for extra mandatory attendees who will be attending the event and perform the same splitting computation since we'd already know the event time causes conflict for the meeting. But yes I agree the initialization is pretty redundant so I'll remove that and also rename the variable to make it more clear! Thanks for catching it.